sonar-db/src/main/java/org/sonar/db/CollationChecker.java

```diff
@@ -0,0 +1,233 @@
+ * As obviously tables must be checked after being created, this component
+ * must not be executed at the same time as {@link DatabaseChecker}.
+ */
+public class CollationChecker implements Startable {
+
+  private final Database db;
+  private final StatementExecutor statementExecutor;
+
+  public CollationChecker(Database db) {
+    this(db, new StatementExecutor());
+  }
+
+  @VisibleForTesting
+  CollationChecker(Database db, StatementExecutor statementExecutor) {
+    this.db = db;
+    this.statementExecutor = statementExecutor;
+  }
+
+  @Override
+  public void start() {
+    try {
+      Loggers.get(getClass()).info("Verify database collation");
+      check();
+    } catch (SQLException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Override
+  public void stop() {
+    // nothing to do
+  }
+
+  private void check() throws SQLException {
+    try (Connection connection = db.getDataSource().getConnection()) {
+      switch (db.getDialect().getId()) {
+        case Oracle.ID:
+          checkOracle(connection);
+          break;
+        case PostgreSql.ID:
+          checkPostgreSql(connection);
+          break;
+        case MySql.ID:
+          checkMySql(connection);
+          break;
+        case MsSql.ID:
+          checkMsSql(connection);
+          break;
+      }
+    }
+  }
+
+  /**
+   * Oracle does not allow to override character set on tables. Only global charset is verified.
+   */
+  private void checkOracle(Connection connection) throws SQLException {
+    String charset = selectSingleCell(connection, "select value  from nls_database_parameters where parameter='NLS_CHARACTERSET'");
+    String sort = selectSingleCell(connection, "select value from nls_database_parameters where parameter='NLS_SORT'");
+    if (!containsIgnoreCase(charset, "UTF8") || !"BINARY".equalsIgnoreCase(sort)) {
+      throw MessageException.of(format("Oracle must be have UTF8 charset and BINARY sort. NLS_CHARACTERSET is %s and NLS_SORT is %s.", charset, sort));
+    }
+  }
+
+  /**
+   * PostgreSQL does not support case-insensitive collations. Only character set must be verified.
+   */
+  private void checkPostgreSql(Connection connection) throws SQLException {
+    // Character set is defined globally and can be overridden on each column.
+    // This request returns all VARCHAR columns. Collation may be empty.
+    // Examples:
+    // issues | key | ''
+    // projects | name | utf8
+    List<String[]> rows = select(connection, "select table_name, column_name, collation_name " +
+      "from information_schema.columns " +
+      "where table_schema='public' " +
+      "and udt_name='varchar' " +
+      "order by table_name, column_name", 3);
+    boolean mustCheckGlobalCollation = false;
+    List<String> errors = new ArrayList<>();
+    for (String[] row : rows) {
+      if (StringUtils.isBlank(row[2])) {
+        mustCheckGlobalCollation = true;
+      } else if (!containsIgnoreCase(row[2], "utf8")) {
```

Comment:
for "utf8" and "UTF8", since you compare ignoring case, you could use the same constant in  `checkOracle`, `checkPostgreSql` and `checkMySql`